Bio/SeqIO/GfaIO.py

```diff
@@ -0,0 +1,89 @@
+"""Bio.SeqIO support for the Graphical Fragment Assembly format."""
+
+import warnings
+import hashlib
+
+from Bio import BiopythonWarning
+from Bio.File import as_handle
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+
+def _check_tags(seq, tags):
+    """Check a segment line's tags for inconsistencies (PRIVATE)."""
+    for tag in tags:
+        if tag[:2] == "LN":
+            # Sequence length
+            if int(tag[5:]) != len(seq):
+                warnings.warn(
+                    f"Segment line has incorrect length. Expected {tag[5:]} but got {len(seq)}.",
+                    BiopythonWarning,
+                )
+        if tag[:2] == "SH":
+            # SHA256 checksum
+            checksum = hashlib.sha256(str(seq).encode()).hexdigest()
+            if checksum.upper() != tag[5:]:
+                warnings.warn(
+                    f"Segment line has incorrect checksum. Expected {tag[5:]} but got {checksum}.",
+                    BiopythonWarning,
+                )
+
+
+def GfaIterator(source):
+    """Parser for GFA 1.x files.
+
+    Documentation: https://gfa-spec.github.io/GFA-spec/GFA1.html
+    """
+    with as_handle(source) as handle:
+        for line in handle:
+            if line == "\n":
+                warnings.warn("GFA data has a blank line.", BiopythonWarning)
```

Comment:
Maybe add a ``continue`` line here? Otherwise won't it fail at ``fields[0]`` a few lines down?

Reply:
Oops, yep you're right. I'll add a commit to fix this later today.